auction-app-frontend/src/components/Home/HomeProductsTabContainer/HomeProductsTabContainer.js

```diff
@@ -1,81 +1,134 @@
-        { id: 6, img: productImg2, name: 'Shoes Collection', startPrice: '59.00' },
-        { id: 7, img: productImg9, name: 'Shoes Collection', startPrice: '59.00' },
-        { id: 8, img: productImg12, name: 'Shoes Collection', startPrice: '59.00' },
-    ];
-
-    const [products, setProducts] = useState(newArrivalsProducts);
+    const pageSize = 4;
+
+    const isInitialMount = useRef(true);
+    const [products, setProducts] = useState([]);
+    const [loadedInitialProducts, setLoadedInitialProducts] = useState(false);
+    const [page, setPage] = useState(0);
+    const [isLastPage, setIsLastPage] = useState(false);
     const [activeTab, setActiveTab] = useState({ newArrivals: true, lastChance: false });
+    const [tabLoading, setTabLoading] = useState(false);
+
+    useEffect(() => {
+        setTabLoading(false);
+        if (!isInitialMount.current && !loadedInitialProducts) {
+            setLoadedInitialProducts(true);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [products]);
+
+    useEffect(() => {
+        if (page === 0) {
+            fetchProductsForTab(activeTab, page, pageSize, true);
+        } else {
+            fetchProductsForTab(activeTab, page, pageSize, false);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [page]);
 
-    function setActive(tab) {
-        const tabSelector = {newArrivals: false, lastChance: false};
+    useEffect(() => {
+        if (isInitialMount.current) {
+            isInitialMount.current = false;
+        } else {
+            setTabLoading(true);
+            if (page === 0) {
+                fetchProductsForTab(activeTab, 0, pageSize, true);
+            } else {
+                setPage(0);
+                setIsLastPage(false);
+            }
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [activeTab]);
+
+    function setTabToActive(tab) {
+        const tabSelector = {
+            newArrivals: false,
+            lastChance: false
+        };
         tabSelector[tab] = true;
         setActiveTab(tabSelector);
     }
 
+    function fetchProductsForTab(tab, page, size, tabChanged) {
+        setTimeout(() => {
+            if (isLastPage) return;
+
+            let getProducts = ProductService.getNewArrivalsProducts;
+            if (tab.lastChance) {
+                getProducts = ProductService.getLastChanceProducts;
+            }
+
+            getProducts(page, size)
+                .then(response => {
+                    setIsLastPage(response.data.currentPage + 1 === response.data.totalPages);
+                    if (tabChanged) {
+                        setProducts(response.data.products);
+                    } else {
+                        setProducts([...products, ...response.data.products]);
+                    }
+                })
+                .catch(err => {
+                    console.log(err);
```

Comment:
Remove `console.log`. We can show error using Alert component you already defined.